daemon/daemon.go

```diff
@@ -141,6 +146,54 @@ func (daemon *Daemon) Features() *map[string]bool {
 	return &daemon.configStore.Features
 }
 
+// NewResolveOptionsFunc returns a call back function to resolve "registry-mirrors" and
+// "insecure-registries" for buildkit
+func (daemon *Daemon) NewResolveOptionsFunc() resolver.ResolveOptionsFunc {
+	return func(ref string) docker.ResolverOptions {
+		var (
+			registryKey = "docker.io"
+			mirrors     = make([]string, len(daemon.configStore.Mirrors))
+			m           = map[string]resolver.RegistryConf{}
+		)
+		// must trim "https://" prefix
+		for i, v := range daemon.configStore.Mirrors {
+			mirrors[i] = strings.Split(v, "//")[1]
```

Comment:
this is going to panic if there is no `//`, which according to https://github.com/moby/moby/blob/master/registry/service_v2.go#L15 could happen.

Reply:
Cleaner to use `strings.Index`, maybe like
```
if idx := strings.Index(v, "://"); idx > 0 {
        v = v[idx+3:]
}
```